Source/WebCore/platform/text/WritingMode.h

```diff
@@ -75,54 +75,44 @@ constexpr inline BlockFlowDirection writingModeToBlockFlowDirection(WritingMode
     return BlockFlowDirection::TopToBottom;
 }
 
-constexpr unsigned makeTextFlowInitalizer(BlockFlowDirection blockFlow, TextDirection direction)
-{
-    return static_cast<unsigned>(blockFlow) << 1 | static_cast<unsigned>(direction);
-}
-
 // Define the text flow in terms of the writing mode and the text direction. The first
 // part is the block flow direction and the second part is the inline base direction.
-enum TextFlow {
-    InlineEastBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::LTR),
-    InlineWestBlockSouth = makeTextFlowInitalizer(BlockFlowDirection::TopToBottom, TextDirection::RTL),
-    InlineEastBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::LTR),
-    InlineWestBlockNorth = makeTextFlowInitalizer(BlockFlowDirection::BottomToTop, TextDirection::RTL),
-    InlineSouthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::LTR),
-    InlineSouthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::LeftToRight, TextDirection::RTL),
-    InlineNorthBlockEast = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::LTR),
-    InlineNorthBlockWest = makeTextFlowInitalizer(BlockFlowDirection::RightToLeft, TextDirection::RTL)
+struct TextFlow {
+    BlockFlowDirection blockFlowDirection;
+    TextDirection inlineBaseDirection;
 };
 
 constexpr inline TextFlow makeTextFlow(WritingMode writingMode, TextDirection direction)
 {
-    TextDirection inlineBaseDirection = direction;
+    auto inlineBaseDirection = direction;
+
+    // FIXME: Is this correct?
     if (writingMode == WritingMode::SidewaysLr)
         inlineBaseDirection = direction == TextDirection::RTL ? TextDirection::LTR : TextDirection::RTL;
```

Comment:
This should be a constructor on the TextFlow class.

Reply:
I filed a bug for that https://bugs.webkit.org/show_bug.cgi?id=276028